src/tests/testIoManip.cc

```diff
@@ -0,0 +1,161 @@
+template <typename Integer>
+static std::string
+toStdString(const AsHex<Integer> &manipulator)
+{
+    std::ostringstream os;
+    os << manipulator;
+    return os.str();
+}
+
+void
+TestIoManip::testAsHex()
+{
+    // basic values
+    CPPUNIT_ASSERT_EQUAL(std::string("0"), toStdString(asHex(0)));
+    CPPUNIT_ASSERT_EQUAL(std::string("123abc"), toStdString(asHex(0x123abc)));
+
+    // large values
+    CPPUNIT_ASSERT_EQUAL(std::string("7fffffffffffffff"), toStdString(asHex(std::numeric_limits<int64_t>::max())));
+    CPPUNIT_ASSERT_EQUAL(std::string("ffffffffffffffff"), toStdString(asHex(std::numeric_limits<uint64_t>::max())));
+
+    // negative values
+    // C++ defines printing with std::hex in terms of calling std::printf() with
+    // %x (or %X) conversion specifier; printf(%x) interprets its value argument
+    // as an unsigned integer, making it impossible for std::hex to print
+    // negative values as negative hex integers. AsHex has the same limitation.
+    CPPUNIT_ASSERT_EQUAL(std::string("80000000"), toStdString(asHex(std::numeric_limits<int32_t>::min())));
+
+    // integer and integer-like types that std::ostream formats specially by default
+    CPPUNIT_ASSERT_EQUAL(std::string("0"), toStdString(asHex(false)));
+    CPPUNIT_ASSERT_EQUAL(std::string("1"), toStdString(asHex(true)));
+    CPPUNIT_ASSERT_EQUAL(std::string("5a"), toStdString(asHex('Z')));
+    CPPUNIT_ASSERT_EQUAL(std::string("77"), toStdString(asHex(int8_t(0x77))));
+    CPPUNIT_ASSERT_EQUAL(std::string("ff"), toStdString(asHex(uint8_t(0xFF))));
+
+    // other interesting integer-like types we may want to print
+    enum { enumValue = 0xABCD };
+    CPPUNIT_ASSERT_EQUAL(std::string("abcd"), toStdString(asHex(enumValue)));
+    struct { uint8_t bitField:2; } s;
+    s.bitField = 3; // TODO: Convert to default initializer after switching to C++20.
+    CPPUNIT_ASSERT_EQUAL(std::string("3"), toStdString(asHex(s.bitField)));
+
+    // padding with zeros works
+    CPPUNIT_ASSERT_EQUAL(std::string("1"), toStdString(asHex(1).minDigits(1)));
+    CPPUNIT_ASSERT_EQUAL(std::string("01"), toStdString(asHex(1).minDigits(2)));
+    CPPUNIT_ASSERT_EQUAL(std::string("001"), toStdString(asHex(1).minDigits(3)));
+
+    // padding with zeros works even for zero values
+    CPPUNIT_ASSERT_EQUAL(std::string("0000"), toStdString(asHex(0).minDigits(4)));
+
+    // minDigits() does not truncate
+    CPPUNIT_ASSERT_EQUAL(std::string("1"), toStdString(asHex(0x1).minDigits(0)));
+    CPPUNIT_ASSERT_EQUAL(std::string("12"), toStdString(asHex(0x12).minDigits(1)));
+    CPPUNIT_ASSERT_EQUAL(std::string("123"), toStdString(asHex(0x123).minDigits(2)));
+
+    // upperCase() forces uppercase
+    CPPUNIT_ASSERT_EQUAL(std::string("A"), toStdString(asHex(0xA).upperCase()));
+    CPPUNIT_ASSERT_EQUAL(std::string("1A2B"), toStdString(asHex(0x1a2b).upperCase(true)));
+
+    std::ostringstream ss;
+
+    // upperCase(false) forces lowercase
```

Comment:
Why is this test case significantly different from the earlier tests of `true` ?
Surely we do not need this special `ss` variable management.

Reply:
> Why is this test case significantly different from the earlier tests of `true` ?

The earlier tests could use and, hence, used default std::ostream flags (including lowercase). This test (and the other ones that use `ss`) cannot use those defaults and, hence, need an explicit stream object to set the flags on.

> Surely we do not need this special `ss` variable management.

If you see a good way to remove `ss` while still testing what is being tested, please post a specific suggestion.